fineract-provider/src/main/java/org/apache/fineract/cob/loan/FetchAndLockLoanTasklet.java

```diff
@@ -87,10 +92,21 @@ public RepeatStatus execute(@NotNull StepContribution contribution, @NotNull Chu
         return RepeatStatus.FINISHED;
     }
 
-    private void applySoftLock(List<Long> alreadySoftLockedAccounts) {
-        for (Long loanId : alreadySoftLockedAccounts) {
-            LoanAccountLock loanAccountLock = new LoanAccountLock(loanId, LockOwner.LOAN_COB_PARTITIONING);
-            loanAccountLockRepository.save(loanAccountLock);
-        }
+    private void applySoftLock(List<Long> accountsToLock) {
+        LocalDate cobBusinessDate = ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE);
+        jdbcTemplate.batchUpdate("""
+                    INSERT INTO m_loan_account_locks (loan_id, version, lock_owner, lock_placed_on, lock_placed_on_cob_business_date)
+                    VALUES (?, ?, ?, ?, ?)
+                """, accountsToLock, getInClauseParameterSizeLimit(), (ps, id) -> {
```

Comment:
Hi Arnold, Did we compared execution time with @Modifying + @Query on repository method also for this optimization.

Reply:
With that approach, you can't do JDBC level batching. This is the only way to do so.